### PR TITLE
Feature/SOF-7193: Strain matching procedure

### PR DESCRIPTION
**Pull request by VsevolodX**
Goal is to closely copy tutorial by QuantumATK https://docs.quantumatk.com/technicalnotes/interface_builder/interface_builder.html

### CODE REVIEW

File: other/python_transformations/2_imports.py [hunk 1]
```diff
@@ -4,39 +4,43 @@
...
+        """
+        is_url = pkg.startswith("http://") or pkg.startswith("https://")
+        are_dependencies_installed = not is_url
+        await micropip.install(pkg, deps=are_dependencies_installed)
+        # Extract package name for printing
+        pkg_name = pkg.split("/")[-1].split("-")[0] if is_url else pkg.split("==")[0]
+        print(f"Installed {pkg_name}")
+
+    for package in packages:
+        await install_package(package)
```
Comment (timurbazhirov) [inline other/python_transformations/2_imports.py:43]: What happened here??

File: other/python_transformations/4_strain_matching.py [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+    # Connect the hover event
+    fig.canvas.mpl_connect("motion_notify_event", hover)
+
+    # Set the scale and labels
+    plt.xscale("log")
+    plt.yscale("log")
+    plt.xlabel(strain_mode)
+    plt.ylabel("number of atoms")
+
+    plt.show()
```
Comment (timurbazhirov) [inline other/python_transformations/4_strain_matching.py:433]: Move the plotting portion to a subfunction (lines 383-to here)

File: other/python_transformations/4_strain_matching.py [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+LAYER_THICKNESS = 1
+
+# Select distance between layers
+DISTANCE = 3.0
+
+# Select parameters for the ZSL algorithm
+MAX_AREA = 400
+MAX_AREA_TOL = 0.09
+MAX_LENGTH_TOL = 0.03
+MAX_ANGLE_TOL = 0.01
```
Comment (timurbazhirov) [inline other/python_transformations/4_strain_matching.py:43]: Move the settings above imports, like in the existing simple interface example

File: other/python_transformations/4_strain_matching.py [hunk 4]
```diff
@@ -0,0 +1,444 @@
...
+        self.film_structure = film_structure
+        self.film_miller = film_miller
+        self.substrate_miller = substrate_miller
+        self.zslgen = zslgen or ZSLGenerator(bidirectional=True)
+
+        self._find_matches()
+        self._find_terminations()
+
+    def _find_matches(self) -> None:
+        """Finds and stores the ZSL matches."""
```
[the remark this answers is not in the document]
Comment (VsevolodX) [reply other/python_transformations/4_strain_matching.py:143]: Ok. It came from Pymatgen directly.

**Comment by VsevolodX** [comment]
image

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**